homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,222 @@
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_MAIN_USER, CONF_USERS, DOMAIN
+
+PLACEHOLDERS = {"api_account_url": "https://www.last.fm/api/account/create"}
+
+CONFIG_SCHEMA: vol.Schema = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_MAIN_USER): str,
+    }
+)
+
+
+def get_lastfm_user(api_key: str, username: str) -> User:
+    """Get lastFM User."""
+    return LastFMNetwork(api_key=api_key).get_user(username)
+
+
+def validate_lastfm_user(user: User) -> dict[str, str]:
+    """Return error if the user is not correct. None if it is correct."""
+    errors = {}
+    try:
+        user.get_playcount()
+    except WSError as error:
+        if error.details == "User not found":
+            errors["base"] = "invalid_account"
+        elif (
+            error.details
+            == "Invalid API key - You must be granted a valid key by last.fm"
+        ):
+            errors["base"] = "invalid_auth"
+        else:
+            errors["base"] = "unknown"
+    except Exception:  # pylint:disable=broad-except
+        errors["base"] = "unknown"
+    return errors
+
+
+class LastFmConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow handler for LastFm."""
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> LastFmOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return LastFmOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize user input."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.data = user_input.copy()
+            main_user = get_lastfm_user(
+                self.data[CONF_API_KEY], self.data[CONF_MAIN_USER]
+            )
+            errors = validate_lastfm_user(main_user)
+            if not errors:
+                return await self.async_step_friends()
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+            description_placeholders=PLACEHOLDERS,
+            data_schema=self.add_suggested_values_to_schema(CONFIG_SCHEMA, user_input),
+        )
+
+    async def async_step_friends(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Form to select other users and friends."""
+        errors = {}
+        if user_input is not None:
+            valid_users = []
+            for username in user_input[CONF_USERS]:
+                lastfm_user = get_lastfm_user(self.data[CONF_API_KEY], username)
+                lastfm_errors = validate_lastfm_user(lastfm_user)
+                if lastfm_errors:
+                    errors = lastfm_errors
+                else:
+                    valid_users.append(username)
```

Comment:
```suggestion
                errors = validate_lastfm_user(lastfm_user)
                if not errors:
                    user_input.pop(username)
```

Reply:
Euh can you pop items you're iterating over?